src/components/Cluster/ClusterDetail/ClusterDetail.js

```diff
@@ -1,24 +1,61 @@
 import { push } from 'connected-react-router';
 import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
-import PropTypes from 'prop-types';
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Breadcrumb } from 'react-breadcrumbs';
-import { connect, useDispatch } from 'react-redux';
-import { Route, Switch } from 'react-router-dom';
-import { selectTargetRelease } from 'selectors/clusterSelectors';
+import { useDispatch, useSelector } from 'react-redux';
+import { Route, Switch, useRouteMatch } from 'react-router-dom';
+import { getUserIsAdmin } from 'selectors/authSelectors';
+import {
+  selectClusterById,
+  selectTargetRelease,
+} from 'selectors/clusterSelectors';
 import { getAllReleases } from 'selectors/releaseSelectors';
 import { AppRoutes } from 'shared/constants/routes';
 
 import GettingStarted from '../../GettingStarted/GettingStarted';
 import ClusterDetailView from './ClusterDetailView';
 
-const ClusterDetail = ({ match, cluster, clusterId, ...rest }) => {
+const ClusterDetail = () => {
   const dispatch = useDispatch();
 
+  const match = useRouteMatch();
+  const clusterID = match.params.clusterId;
+  const orgID = match.params.orgId;
+
+  const cluster = useSelector((state) => selectClusterById(state, clusterID));
+  const allReleases = useSelector((state) => getAllReleases(state) || {});
+  const release = allReleases[cluster?.release_version] ?? null;
+  const isV5Cluster = useSelector((state) =>
+    state.entities.clusters.v5Clusters.includes(clusterID)
+  );
+  const credentials = useSelector((state) => state.entities.credentials);
+  const catalogs = useSelector((state) => state.entities.catalogs);
+  const nodePools = useSelector((state) => state.entities.nodePools.items);
+  const provider = useSelector((state) => state.main.info.general.provider);
+  const user = useSelector((state) => state.main.loggedInUser);
+  const region = useSelector((state) => state.main.info.general.datacenter);
+  const isAdmin = useSelector(getUserIsAdmin);
+
+  const defaultTargetRelease = useSelector((state) => {
+    const targetReleaseVersion = selectTargetRelease(state, cluster);
+
+    return state.entities.releases.items[targetReleaseVersion] ?? null;
+  });
+  const [targetRelease, setTargetRelease] = useState(defaultTargetRelease);
+  const handleChangeTargetRelease = (newReleaseVersion) => {
```

Comment:
Only needed to move the target release selector inside the component, but moved all selectors, for consistency